qiskit/visualization/interactive/iplot_blochsphere.py

```diff
@@ -15,115 +15,36 @@
                 %matplotlib inline
-
                 qc = QuantumCircuit(2, 2)
                 qc.h(0)
                 qc.cx(0, 1)
                 qc.measure([0, 1], [0, 1])
-
                 backend = BasicAer.get_backend('statevector_simulator')
                 job = execute(qc, backend).result()
                 iplot_bloch_multivector(job.get_statevector(qc))
     """
-
-    # HTML
-    html_template = Template("""
-    <p>
-        <div id="content_$divNumber" style="position: absolute; z-index: 1;">
-            <div id="bloch_$divNumber"></div>
-        </div>
-    </p>
-    """)
-
-    # JavaScript
-    javascript_template = Template("""
-    <script>
-        requirejs.config({
-            paths: {
-                qVisualization: "https://qvisualization.mybluemix.net/q-visualizations"
-            }
-        });
-        data = $data;
-        dataValues = [];
-        for (var i = 0; i < data.length; i++) {
-            // Coordinates
-            var x = data[i][0];
-            var y = data[i][1];
-            var z = data[i][2];
-            var point = {'x': x,
-                        'y': y,
-                        'z': z};
-            dataValues.push(point);
-        }
-
-        require(["qVisualization"], function(qVisualizations) {
-            // Plot figure
-            qVisualizations.plotState("bloch_$divNumber",
-                                      "bloch",
-                                      dataValues,
-                                      $options);
-        });
-    </script>
-    """)
-    rho = _validate_input_state(rho)
-    if figsize is None:
-        options = {}
-    else:
-        options = {'width': figsize[0], 'height': figsize[1]}
-
-    # Process data and execute
-    num = int(np.log2(len(rho)))
-
-    bloch_data = []
-    for i in range(num):
-        pauli_singles = [Pauli.pauli_single(num, i, 'X'), Pauli.pauli_single(num, i, 'Y'),
-                         Pauli.pauli_single(num, i, 'Z')]
-        bloch_state = list(map(lambda x: np.real(np.trace(np.dot(x.to_matrix(), rho))),
-                               pauli_singles))
-        bloch_data.append(bloch_state)
-
-    div_number = str(time.time())
-    div_number = re.sub('[.]', '', div_number)
-
-    html = html_template.substitute({
-        'divNumber': div_number
-    })
-
-    javascript = javascript_template.substitute({
-        'data': bloch_data,
-        'divNumber': div_number,
-        'options': options
-    })
-
-    display(HTML(html + javascript))
+    warnings.warn(
+        "The iplot_bloch_multivector function is deprecated and will be "
+        "removed in a future release. The hosted code this depended on no "
+        "longer exists so this is falling back to use the matplotlib "
+        "visualizations. qiskit.visualization.plot_bloch_multivector should be "
+        "used directly moving forward.", DeprecationWarning, stacklevel=2)
+    plot_bloch_multivector(rho, figsize=figsize)
```

Comment:
We'll need to return the figure here to automatically display the mpl version.

Reply:
Actually we don't want to return here it would be both a change in behavior and result in duplicate images. That will result in duplicate images being shown in jupyter since the matplotlib jupyter hooks would pick up the plot call and draw it and then draw again for the repr for the returned figure (see https://github.com/Qiskit/qiskit-terra/commit/5fbdf5a4352b4724490db01701bfa50d48a0e962). The iplot functions were only designed to work in a jupyter context, so not returning here and relying on `%matplotlib inline` is consistent with expectations.

Reply:
Hmm, maybe something is wrong with my jupyter environment. When I tried on the branch, the MPL image wasn't displayed.
<img width="1107" alt="image" src="https://user-images.githubusercontent.com/2241698/84082601-1fbb6580-a9ae-11ea-9939-06e931cd8312.png">

Reply:
Oh, yep you're totally right. We're in the jupyter context so the figures are being closed already inside the matplotlib functions automatically so yeah we need to return them to get it displayed. I'll fix this now, for some reason I was thinking this wasn't in jupyter despite it explicitly only being in jupyter...

Reply:
ok done in: https://github.com/Qiskit/qiskit-terra/pull/4438/commits/6dcff3befb08243ac40ce70958188019cbc3988e